File: Core/SmartGrid/Models/Call.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SmartGrid.Models
{
    public class Call
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        public string Reason { get; set; }
        [Required]
        public string Comment { get; set; }
        [Required]
        public string Hazard { get; set; }
        public string Address { get; set; }

        //public Incident Incident { get; set; }
    }
}

File: clientside/src/app/entities/call.ts
export class Call {
    reason: string;
    comment: string;
    hazard: string;
    address: string;

    constructor(reason: string, comment: string = '', hazard: string, address: string){
        this.reason = reason;
        this.comment = comment;
        this.hazard = hazard;
        this.address = address;
    }
}
